### ISSUE TITLE
Triple quotes appear to break folding

### ISSUE THREAD

**Issue by billylanchantin** [elixir-lsp/vscode-elixir-ls#170]
### Environment

* Elixir & Erlang versions (elixir --version):
    ```
    Erlang/OTP 23 [erts-11.1.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

    Elixir 1.11.1 (compiled with Erlang/OTP 23)
    ```
* Operating system: `MacOS Mojave 10.14.6`
* Editor or IDE name (e.g. Emacs/VSCode): `VSCode`
* Editor Plugin/LSP Client name: `jakebecker.elixir-ls 0.6.4`

### Troubleshooting

- [x] Restart your editor (which will restart ElixirLS) sometimes fixes issues
- [x] Stop your editor, remove the entire `.elixir_ls` directory, then restart your editor
  * NOTE: This will cause you to have to re-run the dialyzer build for your project

### Logs

Not reporting a failure.

## Description

Triple quotes appear to break folding.

When triple quotes are used inside a function at the first level of indentation, that function can no longer be folded (note the lack of a caret on line `10`):

<img width="300" alt="Screen Shot 2021-02-08 at 11 03 12 AM" src="https://user-images.githubusercontent.com/10274508/107245588-4d552480-69fd-11eb-92d3-a08c842bc0a4.png">

When triple quotes are used inside a function as part of a variable, that function's fold ends at the end of the triple quotes:

<table>
<tr valign="top">
<td>
<img width="300" alt="Screen Shot 2021-02-08 at 11 10 38 AM" src="https://user-images.githubusercontent.com/10274508/107246708-83df6f00-69fe-11eb-8b59-1b9026f7a94e.png">
</td>
<td>
<img width="300" alt="Screen Shot 2021-02-08 at 11 12 02 AM" src="https://user-images.githubusercontent.com/10274508/107246706-8346d880-69fe-11eb-8d29-15f6af195801.png">
</td>
</tr>
</table>

## Steps to reproduce

1. `mix new tmp && cd tmp`
2. Open `lib/tmp.ex` in text editor
3. Add a `"""`-pair to a function

## Notes

Please let me know if this was the wrong place to raise this issue!

**Comment by axelson** [comment]
Hi :wave:,

Currently that folding is provided by vscode-elixir-ls so I have transferred your issue to that repository. Specific folding was introduced in https://github.com/elixir-lsp/vscode-elixir-ls/pull/157
Unfortunately from what I can tell it appears that the folding configuration of VSCode does not support specifying which start markers match with which end markers: https://code.visualstudio.com/api/language-extensions/language-configuration-guide#folding

Ideally in the future this would be handled by ElixirLS implementing `textDocument/foldingRange` which has an open issue at https://github.com/elixir-lsp/elixir-ls/issues/358 but I don't believe that it is being actively worked on.

**Comment by billylanchantin** [comment]
@axelson Thanks for the transfer! I poked around both repos (and searched for "fold" in both), but couldn't tell which one was responsible for the feature.

> Ideally in the future this would be handled by ElixirLS implementing textDocument/foldingRange which has an open issue at elixir-lsp/elixir-ls#358 but I don't believe that it is being actively worked on.

Ah that's unfortunate. I love to hit `⌘K2` when working on with `.ex` files. I'll take a look at the progress there but I don't know anything about Language Servers, so I don't know if I'd be able to help out.

**Comment by axelson** [comment]
This is fixed by https://github.com/elixir-lsp/elixir-ls/pull/492 and will be released with 0.7.0 soon
